ext/phar/dirstream.c

```diff
@@ -592,13 +591,12 @@ int phar_wrapper_rmdir(php_stream_wrapper *wrapper, const char *url, int options
 	}
 
 	if (!entry->is_deleted) {
-		for (zend_hash_internal_pointer_reset(&phar->manifest);
-			HASH_KEY_NON_EXISTENT != zend_hash_get_current_key(&phar->manifest, &str_key, &unused);
-			zend_hash_move_forward(&phar->manifest)
-		) {
-			if (ZSTR_LEN(str_key) > path_len &&
-				memcmp(ZSTR_VAL(str_key), ZSTR_VAL(resource->path)+1, path_len) == 0 &&
-				IS_SLASH(ZSTR_VAL(str_key)[path_len])) {
+		ZEND_HASH_FOREACH_STR_KEY(&phar->manifest, str_key) {
```

Comment:
Can't this be `ZEND_HASH_MAP_FOREACH_STR_KEY` ? I see there's already other code that assumes it's a map type array, which makes sense to me as well.

Reply:
TIL about this macro

Reply:
Just gonna say also: You may only use this if you know 100% sure it's not a packed array

Reply:
Well, it seems to be passing all the tests locally, and I don't see how it could be packed.